From d6e60b6703bc439d5b15307fb02a9b9ead2b86ae Mon Sep 17 00:00:00 2001
From: Jason Wessel <jason.wessel@windriver.com>
Date: Wed, 26 Jun 2019 13:58:50 -0700
Subject: [PATCH 1/9] fsck: Throw and error and return non-zero for
 non-verified commits

After the corruption has been fixed with "ostree fsck -a --delete", a
second run of the "ostree fsck" command will print X partial commits
not verified and exit with a zero.

The zero exit code makes it hard to detect if a repair operation needs
to be run, it is easier to check an exit code than try to parse the
text that is emitted from the ostree fsck.

Upstream-Status: Inappropriate [LAT specific]

Signed-off-by: Jason Wessel <jason.wessel@windriver.com>
Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
Signed-off-by: Zhang Xiao <xiao.zhang@windriver.com>
---
 src/ostree/ot-builtin-fsck.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ostree/ot-builtin-fsck.c b/src/ostree/ot-builtin-fsck.c
index 8e3266a1..58cb97e3 100644
--- a/src/ostree/ot-builtin-fsck.c
+++ b/src/ostree/ot-builtin-fsck.c
@@ -448,7 +448,7 @@ ostree_builtin_fsck (int argc, char **argv, OstreeCommandInvocation *invocation,
     }
   else if (n_partial > 0)
     {
-      g_print ("%u partial commits not verified\n", n_partial);
+      return glnx_throw (error, "%u partial commits not verified", n_partial);
     }
 
   if (found_corruption)
-- 
2.34.1

